sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -849,15 +1114,64 @@ object GpuAggFinalPassIterator {
     val opTime = metrics.opTime
     sbIter.map { sb =>
       NvtxIdWithMetrics(NvtxRegistry.FINALIZE_AGG, aggTime, opTime) {
-        val finalBatch = boundExpressions.boundFinalProjections.map { exprs =>
+        val finalBatch = boundExpressions.boundFinalProjections.map { case (exprs, advFns) =>
           SpillableColumnarBatch(
-            GpuProjectExec.projectAndCloseWithRetrySingleBatch(sb, exprs),
+            processAdvancedAggsAndClose(
+              GpuProjectExec.projectAndCloseWithRetrySingleBatch(sb, exprs), advFns),
             SpillPriorities.ACTIVE_BATCHING_PRIORITY)
         }.getOrElse(sb)
         reorderFinalBatch(finalBatch, boundExpressions, metrics)
       }
     }
   }
+
+  /**
+   * The input batch "inputCb" contains the final columns of the build-in aggregates
+   * and the argument columns of the advanced aggregates.
+   *
+   * This function extracts the argument columns from the input batch, and perform
+   * the final "postProcess" action, then insert the result columns into the output
+   * batch at the correct position for each advanced aggregate.
+   * It also passes through the final columns of build-in aggregates to the output
+   * batch.
+   */
+  private[this] def processAdvancedAggsAndClose(inputCb: ColumnarBatch,
+      processOps: Seq[AggregateUtils.AdvancedAggHandler]): ColumnarBatch = {
+    if (processOps.isEmpty) {
```

Comment:
IMHO, performing the path selection before calling the AdvAgg-specific function would make the default path more readable, since developers who aren't concerned with AdvAgg won't need to navigate into that function.